mender.go

```diff
@@ -539,9 +566,61 @@ func reauthorize(m *mender) func() (client.AuthToken, error) {
 	}
 }
 
+// setNextServer is an iterator like function that cycles through the
+// list of available servers in mender.menderConfig.Servers and puts
+// the server as "current active", i.e. fills out the fields
+// mender.menderConfig.ServerURL and mender.menderConfig.TenantToken
+// with the corresponding fields in the server list.
+func setNextServer(m *mender) func() string {
```

Comment:
Confusing name. It sounds like you're using the `m` argument to set some state somewhere, not returning anything. I would perhaps call it `nextServerIterator` or `nextServerFactory` or something like that.

Reply:
I agree, this function has been through some changes since I gave it the name.

Reply:
+1 `nextServerIterator`